### PR TITLE
Create CI github action

### PR DESCRIPTION
**Pull request by notanthony**
Most basic CI is just to try to see if cone is working, maybe worth trying to ping the api to see if end points work. Test fails rn because cone is actually broken, which I am talking to speakeasy about,

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    # Clone the 'conductorone-sdk-go' repository
+    - name: Clone 'conductorone-sdk-go' repository
+      run: git clone https://github.com/conductorone/conductorone-sdk-go.git
+
+    # Replace the vendored SDK in 'cone' with the checked-out version
+    - name: Replace vendored SDK
+      run: |
+        rm -rf cone/vendor/github.com/conductorone/conductorone-sdk-go
+        cp -r conductorone-sdk-go cone/vendor/github.com/conductorone/conductorone-sdk-go
```
Comment (ggreer) [inline .github/workflows/ci.yml:37]: You might want to use `cp -a` to preserve permissions. I don't know if certain things need to be chmodded executable to work.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,4 +22,4 @@ jobs:
 
       # Change directory to '.ci' and test the code
       - name: Change directory to 'cone'
-        run: go build .ci/test.go
+        run: go build - o cli ./.ci
```
Comment (ggreer) [inline .github/workflows/ci.yml:25]: I think you mean `go build -o cli ./.ci` (no space between the dash and o)

File: go.mod [hunk 3]
```diff
@@ -1,19 +1,27 @@
 module github.com/conductorone/conductorone-sdk-go
 
-go 1.14
+go 1.20
```
Comment (ggreer) [inline go.mod:3]: Minor nitpick: We were using go v1.21 before. Is that not available in github actions yet?

Comment (notanthony) [reply go.mod:3]: Oh it should be, I will use that version

**Pull request merged**